scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -234,12 +246,23 @@ def flash(self, **kwargs) -> None:
 
         # flash image and run application
         dl_file = self.cfg.elf_file if self._use_elf else self.cfg.hex_file
+        if self._use_elf:
+            dl_file = self.cfg.elf_file
+            flash_and_run_args = ["--download", dl_file]
+        elif self.cfg.bin_file is not None and os.path.isfile(self.cfg.bin_file) and \
+            "zephyr_signed" in self.cfg.bin_file:
+            dl_file = self.cfg.bin_file
+            dl_address = f"0x{self._download_address:X}"
+            flash_and_run_args = ["--download", dl_file, dl_address]
+        elif self.cfg.hex_file is not None and os.path.isfile(self.cfg.hex_file):
+            # --user-elf not used and no bin file given, default to hex
+            dl_file = self.cfg.hex_file
+            flash_and_run_args = ["--download", dl_file]
         if dl_file is None:
             raise RuntimeError('cannot flash; no download file was specified')
         elif not os.path.isfile(dl_file):
             raise RuntimeError(f'download file {dl_file} does not exist')
```

Comment:
Please adapt to the following to ease further changes for USB-DFU support

```suggestion
        if self._use_elf:
            dl_file = self.cfg.elf_file
        elif self.cfg.bin_file is not None and os.path.isfile(self.cfg.bin_file) and \
            "zephyr_signed" in self.cfg.bin_file:
            dl_file = self.cfg.bin_file
        elif self.cfg.hex_file is not None and os.path.isfile(self.cfg.hex_file):
            # --user-elf not used and no bin file given, default to hex
            dl_file = self.cfg.hex_file
        if dl_file is None:
            raise RuntimeError('cannot flash; no download file was specified')
        elif not os.path.isfile(dl_file):
            raise RuntimeError(f'download file {dl_file} does not exist')
            
        flash_and_run_args = ["--download", dl_file]
        if self._download_address is not None:
            flash_and_run_args.append(f"0x{self._download_address:X}")
```